codes/c/utils/greedy.h

```diff
@@ -0,0 +1,16 @@
+
+#ifndef _GREEDY_H_
+#define _GREEDY_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define MIN(a, b) (a < b ? a : b)
```

Comment:
I'd be clearer to directly use `a < b ? a : b` in the source code rather than define `MIN`. Same as `MAX`.

Reply:
Moved MIN&MAX macro to source file.